modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestSdkClientCodegen.java

```diff
@@ -300,7 +300,7 @@ public String getTypeDeclaration(Schema p) {
             return getSchemaType(p) + "<utility::string_t, " + getTypeDeclaration(inner) + ">";
         } else if (ModelUtils.isStringSchema(p)
                 || ModelUtils.isDateSchema(p) || ModelUtils.isDateTimeSchema(p)
-                || ModelUtils.isFileSchema(p)
+                || ModelUtils.isFileSchema(p) || ModelUtils.isUUIDSchema(p)
```

Comment:
@Peaches491 thanks again for the PR. Should the `isStringSchema` check at line 301 already cover that (type: string)?
```
    public static boolean isStringSchema(Schema schema) {
        if (schema instanceof StringSchema || SchemaTypeUtil.STRING_TYPE.equals(schema.getType())) {
            return true;
        }
        return false;
    }
```
Ref: https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java#L324-L329

Reply:
I don't believe so, since adding this caused the generated code to be different. Specifically, string parameters with `format: "uuid"` were no longer used a `std::shared_ptr<utilities::string_t>` and instead just used the expected `utilities::string_t`

Reply:
You're right. I did a test and got:
```
         int64_t petId,
-        boost::optional<utility::string_t> name,
+        boost::optional<std::shared_ptr<utility::string_t>> name,
         boost::optional<utility::string_t> status
```